include/stdio.h

```diff
@@ -244,6 +244,11 @@ int    dprintf(int fd, FAR const IPTR char *fmt, ...) printf_like(2, 3);
 int    vdprintf(int fd, FAR const IPTR char *fmt, va_list ap)
        printf_like(2, 0);
 
+/* Custom stream operation fopencookie. */
+
+FAR FILE *fopencookie(FAR void *cookie, FAR const char *mode,
+                      cookie_io_functions_t io_funcs);
+
```

Comment:
It is prohibited to add non-standard interfaces to the OS.  We are purely standards based and non-standard, ad hoc OS interfaces are generally prohibited.

Reply:
it is implemented by *BSD too:
https://man.freebsd.org/cgi/man.cgi?query=fopencookie&apropos=0&sektion=3&manpath=FreeBSD+11-current&format=html

Reply:
Yes, I think the usage in Linux is a justification here although it is not a POSIX interface. Morover this will be helpful in the implementation of `fmemopen` https://pubs.opengroup.org/onlinepubs/9699919799/functions/fmemopen.html (although we can do it without `fopencookie` but it complicates things a bit).